burn-tensor/src/tensor/ops/tensor.rs

```diff
@@ -1075,4 +1075,88 @@ pub trait TensorOps<B: Backend> {
 
         (values, index)
     }
+
+    /// Returns a new tensor with the given dimension narrowed to the given range.
+    ///
+    /// # Arguments
+    ///
+    /// * `dim` - The dimension along which the tensor will be narrowed.
+    /// * `start` - The starting point of the given range.
+    /// * `length` - The ending point of the given range.
+    /// # Panics
+    ///
+    /// - If the dimension is greater than the number of dimensions of the tensor.
+    /// - If the given range exceeds the number of elements on the given dimension.
+    ///
+    /// # Returns
+    ///
+    /// A new tensor with the given dimension narrowed to the given range.
+    fn narrow<const D: usize>(
+        tensor: FloatTensor<B, D>,
+        dim: usize,
+        start: usize,
+        length: usize,
+    ) -> FloatTensor<B, D> {
+        let ranges: Vec<_> = (0..D)
+            .map(|i| {
+                if i == dim {
+                    start..(start + length)
+                } else {
+                    0..Self::shape(&tensor).dims[i]
+                }
+            })
+            .collect();
+
+        let ranges_array: [_; D] = ranges.try_into().unwrap();
+
+        Self::slice(tensor, ranges_array)
+    }
+
+    /// Split the tensor along the given dimension into chunks.
+    ///
+    /// # Arguments
+    ///
+    /// * `tensor` - The tensor.
+    /// * `chunks` - The number of chunks to be produced
+    /// * `times` - The dimension along which the tensor will be split.
+    ///
+    /// # Returns
+    ///
+    /// A vectors of tensors
+    ///
+    fn chunk<const D: usize>(
```

Comment:
It seems you have started your branch before #1006 was merged. 
In #1006 we modified the default chunk algorithm --don't mind the name of the PR which is about fixing the docs, at first I thought the doc was wrong and the code was right, but in the end the doc was right and the code was wrong. 
Could you please adjust the algorithm so it passes the test in #1006 (or in main as #1006 was merged) ?

Reply:
Sure, let me work on that.